File: sdks/go/pkg/models/operations/getretentionpolicyassignmentsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetRetentionPolicyAssignmentsIDPathParams struct {
	RetentionPolicyAssignmentID string `pathParam:"style=simple,explode=false,name=retention_policy_assignment_id"`
}

type GetRetentionPolicyAssignmentsIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetRetentionPolicyAssignmentsIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetRetentionPolicyAssignmentsIDPathParams
	QueryParams GetRetentionPolicyAssignmentsIDQueryParams
}

type GetRetentionPolicyAssignmentsIDResponse struct {
	ClientError               *shared.ClientError
	ContentType               string
	RetentionPolicyAssignment *shared.RetentionPolicyAssignment
	StatusCode                int64
}

File: ts-sdk/src/sdk/models/operations/getretentionpolicyassignmentsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRetentionPolicyAssignmentsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=retention_policy_assignment_id" })
  retentionPolicyAssignmentId: string;
}


export class GetRetentionPolicyAssignmentsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetRetentionPolicyAssignmentsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetRetentionPolicyAssignmentsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetRetentionPolicyAssignmentsIdQueryParams;
}


export class GetRetentionPolicyAssignmentsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  retentionPolicyAssignment?: shared.RetentionPolicyAssignment;

  @SpeakeasyMetadata()
  statusCode: number;
}
